### PR TITLE
feat documentation authorization

### LINKED ISSUE #465: fastify-swagger: protect access of Swagger UI
Hello ! 

I'm using [fastify-swagger ](https://github.com/fastify/fastify-swagger) to generate API documentation. 
It's working fine but I would like to protect the access to Swagger UI with a simple username / password.

Is it possible do to it with fastify-swagger ?

### PR DESCRIPTION
**Pull request by Eomm**
closes #465

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -642,6 +647,32 @@ There are two ways to hide a route from the Swagger UI:
 - Pass `{ hide: true }` to the schema object inside the route declaration.
 - Use the tag declared in `hiddenTag` options property inside the route declaration. Default is `X-HIDDEN`.
 
+<a name="route.uiHooks"></a>
+#### Protect your documentation routes
+
+You can protect your documentation configuring an authentication hook.
```
Comment (zekth) [inline README.md:653]:
```suggestion
You can protect your documentation by configuring an authentication hook.
```

File: README.md [hunk 2]
```diff
@@ -642,6 +647,32 @@ There are two ways to hide a route from the Swagger UI:
...
+Here is an example with the [`fastify-basic-auth`](https://github.com/fastify/fastify-basic-auth) plugin:
```
Comment (zekth) [inline README.md:654]:
```suggestion
Here is an example using [`fastify-basic-auth`](https://github.com/fastify/fastify-basic-auth) plugin:
```

**Review by zekth — CHANGES_REQUESTED** [review]
Minor rewording remarks

File: README.md [hunk 3]
```diff
@@ -642,6 +647,32 @@ There are two ways to hide a route from the Swagger UI:
 - Pass `{ hide: true }` to the schema object inside the route declaration.
 - Use the tag declared in `hiddenTag` options property inside the route declaration. Default is `X-HIDDEN`.
 
+<a name="route.uiHooks"></a>
+#### Protect your documentation routes
+
+You can protect your documentation by configuring an authentication hook.
+Here is an example using [`fastify-basic-auth`](https://github.com/fastify/fastify-basic-auth) plugin:
```
Comment (Fdawgs) [inline README.md:654]:
```suggestion
Here is an example using the [`fastify-basic-auth`](https://github.com/fastify/fastify-basic-auth) plugin:
```

**Comment by mcollina** [comment]
@Eomm could you take a look?

**Review by mcollina — APPROVED** [review]
lgtm

File: lib/routes.js [hunk 4]
```diff
@@ -64,6 +76,7 @@ function fastifySwagger (fastify, opts, done) {
     url: '/uiConfig',
     method: 'GET',
     schema: { hide: true },
+    ...hooks,
```
Comment (climba03003) [inline lib/routes.js:79]: I do not think the UI can works properly when fetching the config require hooks or authentication as the script do not bring the `authorization` header.

Comment (Eomm) [reply lib/routes.js:79]: It does

image

Fetch includes the authorization for the same-origin
https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#sending_a_request_with_credentials_included

Comment (climba03003) [reply lib/routes.js:79]: Oh. I didn't imagine it does.

**Review by climba03003 — APPROVED** [review]
LGTM.

**Comment by mcollina** [comment]
@zekth you good?

**Review by zekth — APPROVED** [review]
lgtm!

**Review by Fdawgs — APPROVED** [review]

**Pull request merged**